olp-cpp-sdk-dataservice-read/src/repositories/DataRepository.h

```diff
@@ -34,27 +34,26 @@ namespace repository {
 
 class DataRepository final {
  public:
-  static DataResponse GetVersionedTile(
-      const client::HRN& catalog, const std::string& layer_id,
-      const TileRequest& request, int64_t version,
-      client::CancellationContext context,
-      const client::OlpClientSettings& settings);
-
-  static DataResponse GetVersionedData(
-      const client::HRN& catalog, const std::string& layer_id, int64_t version,
-      DataRequest data_request, client::CancellationContext context,
-      const client::OlpClientSettings& settings);
-
-  static DataResponse GetVolatileData(
-      const client::HRN& catalog, const std::string& layer_id,
-      DataRequest request, client::CancellationContext context,
-      const client::OlpClientSettings& settings);
-
-  static DataResponse GetBlobData(
-      const client::HRN& catalog, const std::string& layer,
-      const std::string& service, const DataRequest& data_request,
-      client::CancellationContext cancellation_context,
-      const client::OlpClientSettings& settings);
+  DataRepository(client::HRN catalog, client::OlpClientSettings settings);
+
+  DataResponse GetVersionedTile(const std::string& layer_id,
+                                const TileRequest& request, int64_t version,
+                                client::CancellationContext context);
+
+  DataResponse GetVersionedData(const std::string& layer_id,
+                                DataRequest data_request, int64_t version,
+                                client::CancellationContext context);
+
+  DataResponse GetVolatileData(const std::string& layer_id, DataRequest request,
+                               client::CancellationContext context);
+
+  DataResponse GetBlobData(const std::string& layer, const std::string& service,
+                           const DataRequest& data_request,
+                           client::CancellationContext context);
```

Comment:
Sometimes request is passed by const ref, sometimes by value, why it is not consistent?

Reply:
if data request has no data handle GetVersionedData() and GetVolatileData() will modify the request.
I will make it const reference and copy the request inside.